### PR TITLE
add find_best_spot_price.py to close #3

### LINKED ISSUE #3: Finish find_best_spot_price.Py
Make Python code to find cheap regions with two AZs for running ec2 instances

### PR DESCRIPTION
**Pull request by zr2d2**

### CODE REVIEW

File: cheep.py
Comment (duffsterlp) [inline cheep.py:1]: Suggest renaming the file to something more descriptive. One idea is `find_best_spot_price.py`.

File: cheep.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import boto3
+import sys
+
+def get_regions():
+    regions = ['us-west-1','us-west-2','us-east-1','us-east-2','ap-northeast-1','ap-northeast-2','ap-northeast-3','ap-south-1','ap-southeast-1','ap-southeast-2','ca-central-1','eu-central-1','eu-north-1','eu-west-1','eu-west-2','eu-west-3']#,'sa-east-1','us-east-1','us-east-2','us-west-1','us-west-2']
```
Comment (duffsterlp) [inline cheep.py:5]: Suggest removing commented out regions or adding a comment why they're commented out

File: cheep.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    ec2_client = boto3.client('ec2', region_name='us-east-1')
+    try:
+        response = ec2_client.describe_regions()
+        return [region['RegionName'] for region in response['Regions']]
+    except RuntimeError as e:
+        print("Warning: Can't describe_regions. Falling back to default list")
```
Comment (duffsterlp) [inline cheep.py:12]: Suggest including the exception details in the print statement

File: cheep.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import boto3
+import sys
+
+def get_regions():
+    regions = ['us-west-1','us-west-2','us-east-1','us-east-2','ap-northeast-1','ap-northeast-2','ap-northeast-3','ap-south-1','ap-southeast-1','ap-southeast-2','ca-central-1','eu-central-1','eu-north-1','eu-west-1','eu-west-2','eu-west-3']#,'sa-east-1','us-east-1','us-east-2','us-west-1','us-west-2']
+
+    ec2_client = boto3.client('ec2', region_name='us-east-1')
```
Comment (duffsterlp) [inline cheep.py:7]: I'm guessing the region_name here doesn't matter and is just for the sake of pointing the client somewhere to find all the regions. Is that correct?

File: cheep.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        if response['SpotPriceHistory']:
+            for spot_price in response['SpotPriceHistory']:
+                if float(spot_price['SpotPrice']) < best_price:
+                    best_price = float(spot_price['SpotPrice'])
+                    best_az = spot_price['AvailabilityZone']
+
+    return best_az, best_price
+
+if __name__ == "__main__":
+    instance_type = 'm5.xlarge'
```
Comment (duffsterlp) [inline cheep.py:36]: Would it be useful to pass in the instance type as a parameter?

File: cheep.py
Comment (zr2d2) [reply cheep.py:1]: makes sense

File: cheep.py (hunk 3 above)
Comment (zr2d2) [reply cheep.py:7]: yes, it's for the region of the endpoint for the API

File: cheep.py (hunk 4 above)
Comment (zr2d2) [reply cheep.py:36]: ya, sure

**Pull request merged**